clang/lib/AST/ASTContext.cpp

```diff
@@ -3442,6 +3442,13 @@ uint16_t ASTContext::getPointerAuthTypeDiscriminator(QualType T) {
     encodeTypeForFunctionPointerAuth(*this, Out, T);
   } else {
     T = T.getUnqualifiedType();
+    // Drop exception specification from member function pointer type.
+    if (auto *MPT = T->getAs<MemberPointerType>())
+      if (MPT->isMemberFunctionPointer()) {
+        QualType FT =
+            getFunctionTypeWithExceptionSpec(MPT->getPointeeType(), EST_None);
+        T = getMemberPointerType(FT, MPT->getClass());
```

Comment:
Let's try to avoid rebuilding if the type already doesn't have an exception spec.